### DISCUSSION TITLE
Are using auto subdirectories/wildcards and compiling to a single pdf mutually exclusive?

### DISCUSSION THREAD

**Discussion by thadryanjs** [quarto-dev/quarto-cli#6636] [Q&A]
### Description

Hello! 

I made a template for some manuscripts I'll be working on which will have a varying number of sections and subsections, which I planned to compile to a single pdf.

I've structured them like this:

```
├── index.qmd
├── _quarto.yml
├── section0
│   ├── index.qmd
│   ├── subsection00.qmd
│   └── subsection01.qmd
└── section1
    ├── index.qmd
    ├── subsection00.qmd
    └── subsection01.qmd
```

I was initially planning on using the `book` template. Because there will likely be a lot of them being added and removed, I was planning on using wildcards, ie:

```
chapters:
    - section0/*
    - section1/*
```

...but found out that wasn't [possible](https://github.com/quarto-dev/quarto-cli/issues/1917).

I tried the [workaround](https://quarto.org/docs/websites/website-navigation.html#auto-generation) mentioned in the post, using the `website` template, but at the time of this writing, this template [can't produce a single pdf](https://github.com/quarto-dev/quarto-cli/discussions/5782) . The OP is asking about a `book` and this is presented as a workaround but I can't get to work with `book`s, only `website`s proper.

It seems you can either 1) use auto with a `website`, or 2) get a single pdf from a `book`, but not both (or 3, I am completely missing something). The `default` project uses wildcards but, unless I'm mistaken, also will produce a pdf for every source file.

Is there a way to use sub-directories to make a single pdf in an automated way or should I try a new approach/script something?

Cheers!

PS: 
Thanks for working on this project - I've found the tool quite handy.

**Comment by mcanouil** [comment]
Could you share a small self-contained "working" (reproducible) example to work with, _i.e._, a complete Quarto document or a Git repository? Thanks.

You can share a Quarto document using the following syntax, _i.e._, using more backticks than you have in your document (usually four ` ```` `).

`````md
````qmd
---
title: "Reproducible Quarto Document"
format: html
---

This is a reproducible Quarto document using `format: html`.
It is written in Markdown and contains embedded R code.
When you run the code, it will produce a plot.

```{r}
plot(cars)
```

The end.
````
`````

**Reply by thadryanjs** [reply]
Thanks for the reply and certainly. I made a repo [here](https://github.com/thadryanjs/quarto-question/tree/master). Each folder is a complete quarto project. There are comments in the yml files.

If I'm understanding what I've read and tried correctly:

- `default` can use wildcards to get the subdirs automatically, but can't make a single pdf
- `website` can use auto to include the subdirs automatically but can't produce a single pdf
- `book` can produce a single pdf, but can't use wildcards or auto to get the subdirs

Thanks!

**Comment by cderv** [comment]
@thadryanjs I believe `books` and auto feature is tracked here: 
* https://github.com/quarto-dev/quarto-cli/issues/1917

It is still to be implemented

**Reply by thadryanjs** [reply]
OK thank you!

I read this thread but didn't notice it was being used for tracking. I will subscribe and stay tuned :)
